components/centraldashboard/public/components/activity-view.js

```diff
@@ -0,0 +1,113 @@
+                    --sidebar-default-color: #ffffff4f;
+                    --border-color: #f4f4f6;
+                    padding: 1em;
+                }
+                paper-progress {
+                    width: 100%;
+                    --paper-progress-active-color: var(--accent-color)
+                }
+                #list {
+                    overflow-y: scroll;
+                }
+                .activity-row {
+                    background: #fff;
+                    border-top: 1px solid rgba(0,0,0,.12);
+                    box-shadow: 0 3px 3px rgba(0,0,0,.12);
+                    transition: margin .2s cubic-bezier(0.4, 0, 0.2, 1);
+                    font-size: 13px;
+                    padding: 5px;
+                }
+                .activity-row div {
+                    padding: 0 5px;
+                }
+                .event {
+                    flex-basis: 50%;
+                }
+                .event-icon {
+                    height: 18px;
+                    width: 18px;
+                    padding: 0 10px;
+                }
+                .event-icon.build {
+                    transform: scale(-1, 1);
+                    color: rgb(28, 175, 199);
+                }
+                .event-icon.error {
+                    color: rgb(218, 66, 54);
+                }
+                [hidden] {
+                    display: none;
+                    opacity: 0;
+                    pointer-events: none
+                }
+            </style>
+            <iron-ajax auto url="/api/activities" handle-as="json"
+                loading="{{loading}}" on-response="_onResponse">
+            </iron-ajax>
+            <paper-progress indeterminate class="slow"
+                hidden$="[[!loading]]"></paper-progress>
+            <div id="list" role="listbox">
+                <template is="dom-repeat" items="[[activities]]">
+                    <div class="activity-row layout horizontal">
+                        <div class="flex">[[item.formattedTime]]</div>
+                        <div class="flex flex-auto event">
+                            <iron-icon class$="event-icon [[item.icon]]"
+                                icon="[[item.icon]]">
+                            </iron-icon>[[item.event]]
+                        </div>
+                        <div class="flex">[[item.source]]</div>
+                    </div>
+                </template>
+            </div>
+            `;
+    }
+
+    /**
+      * Object describing property-related metadata used by Polymer features
+      */
+    static get properties() {
+        return {
+            loading: Boolean,
+            activities: Array,
+        };
+    }
+
+    /**
+     * Handles the Activities response to set date format and icon.
+     * @param {Event} responseEvent
+     */
+    _onResponse(responseEvent) {
+        const {status, response} = responseEvent.detail;
+        this.activities = [];
+        if (status === 200) {
```

Comment:
Can we invert this condition and decrease code depth:
```js
if (status != 200) return;
```

Reply:
Good call. I think we'll want to add some kind of means for surfacing an request failure as well so that block will be expanded.